task/thermal/src/control.rs

```diff
@@ -77,6 +77,42 @@ impl TemperatureSensor {
     }
 }
 
+/// Represents the indvidual fans in the system
+///
+/// Depending on the system we have diferent numbers of fans structured in
+/// different ways. Not all fans are guaranteed to be there at all times so
+/// their corresponding sensor is an `Option`. We should not read the RPM of
+/// fans which are not present and their PWM should only be driven low.
+#[derive(Copy, Clone)]
+pub struct Fans([Option<SensorId>; bsp::NUM_FANS]);
```

Comment:
Instead of using the `bsp` value here, you could make this take a const generic:

```rust
pub struct Fans<const N: usize>([Option<SensorId>; N]);
```

To my mind, this is cleaner than coupling to the BSP in the `struct` type itself.

(Of course, this means that you'll need to use `Fan<{ bsp::NUM_FANS }>` instead of `Fans` elsewhere...)

Reply:
I think decoupling the type from the BSP is indeed a cleaner, more preferable path.